### PR TITLE
Add doc for Faster-RCNN.

### PR DESCRIPTION
**Pull request by jerrywgz**

### CODE REVIEW

File: fluid/faster_rcnn/README_cn.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+- [数据准备](#数据准备)
+- [模型训练](#模型训练)
+- [参数微调](#参数微调)
+- [模型评估](#模型评估)
+- [模型推断及可视化](#模型推断及可视化)
+- [附录](#附录)
+
+## 安装
+
+在当前目录下运行样例代码需要PadddlePaddle Fluid的v0.13.0或以上的版本。如果你的运行环境中的PaddlePaddle低于此版本，请根据[安装文档](http://www.paddlepaddle.org/documentation/docs/zh/0.15.0/beginners_guide/install/install_doc.html#paddlepaddle)中的说明来更新PaddlePaddle。
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: v0.13.0 版本号不对，得 v.1.0.0

File: fluid/faster_rcnn/README_cn.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+       --scales=800 \
+       --batch_size=8 \
+       --batch_size_per_im=512 \
+       --class_dim=81 \
+       --model_save_dir=output/ \
+       --max_iter=180000 \
+       --learning_rate=0.01 \
+       --padding_minibatch=True
+
+- 通过设置export CUDA\_VISIBLE\_DEVICES=0,1指定GPU训练。
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: export CUDA\_VISIBLE\_DEVICES=0,1,2,3,4,5,6,7指定8卡GPU训练。

File: fluid/faster_rcnn/README_cn.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    python train.py \
+       --max_size=1333 \
+       --scales=800 \
+       --batch_size=8 \
+       --batch_size_per_im=512 \
+       --class_dim=81 \
+       --model_save_dir=output/ \
+       --max_iter=180000 \
+       --learning_rate=0.01 \
+       --padding_minibatch=True
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: 一些默认的参数可以省略

File: fluid/faster_rcnn/README_cn.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+*  采用momentum优化算法训练Faster RCNN，momentum=0.9。
+*  权重衰减系数为0.0001，前500轮学习率从0.00333线性增加至0.01。在120000，160000轮时使用0.1,0.01乘子进行学习率衰减，最大训练180000轮。
+*  非基础卷积层卷积bias学习率为整体学习率2倍。
+*  基础卷积层中，affine_layers参数不更新，res2层参数不更新。
+*  使用Nvidia Tesla V100 8卡并行，总共训练时长大约40小时。
+
+## 参数微调
+
+参数微调是指在特定任务上微调已训练模型的参数。通过初始化`pretrained_model`，微调一个模型可以采用如下的命令：
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: 加载初始化模型需要在模型训练那部分说明吧

File: fluid/faster_rcnn/README_cn.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+下图为模型可视化预测结果：
+<p align="center">
+<img src="image/000000000139.jpg" height=300 width=400 hspace='10'/>
+<img src="image/000000127517.jpg" height=300 width=400 hspace='10'/>
+<img src="image/000000203864.jpg" height=300 width=400 hspace='10'/>
+<img src="image/000000515077.jpg" height=300 width=400 hspace='10'/> <br />
+Faster RCNN 预测可视化
+</p>
+
+## 附录
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: 我觉得还是不用加 "模型调试" 这一章节了。

加这里不够系统， Python API也有变化，比如下面 'get_var' 被隐藏了，没有暴露给用户。 觉得暂且不用加。

File: fluid/faster_rcnn/README_cn.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+- 通过设置export CUDA\_VISIBLE\_DEVICES=0,1指定GPU训练。
+- 可选参数见：
+
+    python train.py --help
+
+**数据读取器说明：** 数据读取器定义在reader.py中。所有图像将短边等比例缩放至`scales`，若长边大于`max_size`, 则再次将长边等比例缩放至`max_iter`。在训练阶段，对图像采用水平翻转。支持将同一个batch内的图像padding为相同尺寸。
+
+**模型设置：**
+
+* roi_pool为average pooling。
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: roi_pool为average pooling 这句不太懂。 是说想： 使用RoIPooling？

File: fluid/faster_rcnn/README_cn.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+## 简介
+
+[Faster Rcnn](https://arxiv.org/abs/1506.01497) 是典型的两阶段目标检测器。如下图所示，整体网络可以分为4个主要内容：
+<p align="center">
+<img src="image/Faster_RCNN.jpg" height=400 width=400 hspace='10'/> <br />
+Faster RCNN 目标检测模型
+</p>
+
+1. 基础卷积层。作为一种卷积神经网络目标检测方法，Faster RCNN首先使用一组基础的卷积网络提取图像的feature maps。feature maps被后续RPN层和全连接层共享。本示例采用[ResNet-50](https://arxiv.org/abs/1512.03385)作为基础卷积层。
+2. 区域生成网络(RPN)。RPN网络用于生成候选区域(proposals)。该层通过一组固定的尺寸和比例得到一组锚点(anchors), 通过softmax判断anchors属于前景或者背景，在利用box regression修正anchors获得精确的proposals。
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: box regression翻译成中文？ 需要正式一些。

File: fluid/faster_rcnn/README_cn.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+3. RoI池化。该层收集输入的feature maps和proposals，将proposals映射到feature maps中并池化为统一大小的proposal feature maps，送入全连接层判定目标类别。
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: feature maps -> 特征图？

File: fluid/faster_rcnn/README_cn.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+        --pretrained_mode=${path_to_pretrain_model} \
+        --batch_size=1 \
+        --nms_threshold=0.5 \
+        --score_threshold=0.05
+
+下图为模型评估结果：
+<p align="center">
+<img src="image/mAP.jpg" height=500 width=650 hspace='10'/> <br />
+Faster RCNN mAP
+</p>
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md:116]: 可以再对结果加一个表格。

File: fluid/faster_rcnn/README.md [hunk 10]
```diff
@@ -66,7 +64,7 @@ After data preparation, one can start the training step by:
 
 **model configuration:**
 
-* Roi_pool layer takes average pooling.
+* Use RoIPolling.
```
Comment (qingqing01) [inline fluid/faster_rcnn/README.md]: RoIPolling -> RoIPooling

File: fluid/faster_rcnn/README.md [hunk 11]
```diff
@@ -128,6 +124,13 @@ Evalutaion result is shown as below:
 <p align="center">
 <img src="image/mAP.jpg" height=500 width=650 hspace='10'/> <br />
 Faster RCNN mAP
+
+| Model                    | mAP     |
+| :------------------------------ | -------:     |
+| detectron                 | 0.315    |
+| fluid no padding         | 0.308     |
```
Comment (qingqing01) [inline fluid/faster_rcnn/README.md]: fluid -> Fluid

File: fluid/faster_rcnn/README.md [hunk 12]
```diff
@@ -128,6 +124,13 @@ Evalutaion result is shown as below:
 <p align="center">
 <img src="image/mAP.jpg" height=500 width=650 hspace='10'/> <br />
 Faster RCNN mAP
+
+| Model                    | mAP     |
+| :------------------------------ | -------:     |
```
Comment (qingqing01) [inline fluid/faster_rcnn/README.md]: 再加2栏： batch size, max_iter ?

File: fluid/faster_rcnn/README_cn.md [hunk 13]
```diff
@@ -90,9 +88,7 @@ Faster RCNN 训练loss
         --pretrained_model=${path_to_pretrain_model} \
         --batch_size= 8\
         --model_save_dir=output/ \
-        --class_dim=81 \
-        --max_iter=180000 \
-        --learning_rate=0.01 \
+        --padding_minibatch=True
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: padding_minibatch觉得用默认的Flase就行吧。

File: fluid/faster_rcnn/README_cn.md [hunk 14]
```diff
@@ -122,6 +118,13 @@ Faster RCNN 训练loss
 <p align="center">
 <img src="image/mAP.jpg" height=500 width=650 hspace='10'/> <br />
 Faster RCNN mAP
+
+| 模型                    | mAP     |
+| :------------------------------ | -------:     |
+| detectron                 | 0.315    |
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: detectron -> Detectron

File: fluid/faster_rcnn/README_cn.md [hunk 15]
```diff
@@ -122,6 +118,13 @@ Faster RCNN 训练loss
...
+| fluid no padding         | 0.308     |
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: fluid -> Fluid

File: fluid/faster_rcnn/README.md (hunk 10 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README.md]: Done

File: fluid/faster_rcnn/README.md (hunk 12 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README.md]: Done

File: fluid/faster_rcnn/README.md (hunk 11 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README.md]: Done

File: fluid/faster_rcnn/README_cn.md (hunk 3 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README_cn.md]: Done

File: fluid/faster_rcnn/README_cn.md (hunk 13 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README_cn.md]: Done

File: fluid/faster_rcnn/README_cn.md (hunk 14 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README_cn.md]: Done

File: fluid/faster_rcnn/README_cn.md (hunk 15 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README_cn.md]: Done

File: fluid/faster_rcnn/README_cn.md (hunk 9 above)
Comment (jerrywgz) [reply fluid/faster_rcnn/README_cn.md:116]: Done

File: fluid/faster_rcnn/README_cn.md [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+
+*  采用momentum优化算法训练Faster RCNN，momentum=0.9。
+*  权重衰减系数为0.0001，前500轮学习率从0.00333线性增加至0.01。在120000，160000轮时使用0.1,0.01乘子进行学习率衰减，最大训练180000轮。
+*  非基础卷积层卷积bias学习率为整体学习率2倍。
+*  基础卷积层中，affine_layers参数不更新，res2层参数不更新。
+*  使用Nvidia Tesla V100 8卡并行，总共训练时长大约40小时。
+
+## 参数微调
+
+参数微调是指在特定任务上微调已训练模型的参数。通过初始化`pretrained_model`，微调一个模型可以采用如下的命令：
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md]: 这个模型不加载初始化模型会出NaN， 如何下载初始化模型、加载训练需要移到训练部分。

**Review by qingqing01 — APPROVED** [review]

File: fluid/faster_rcnn/README_cn.md [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+<img src="image/mAP.jpg" height=500 width=650 hspace='10'/> <br />
+Faster RCNN mAP
+</p>
+
+| 模型                    | 批量大小     | 迭代次数        | mAP  |
+| :------------------------------ | :------------:    | :------------------:    |------: |
+| Detectron                 | 8            |    180000        | 0.315 |
+| Fluid all padding         | 8            |    180000        | 0.308 |
+| Fluid minibatch padding | 8            |    180000        | 0.314 |
+| Fluid no padding            |6            |    240000        | 0.317 |
```
Comment (qingqing01) [inline fluid/faster_rcnn/README_cn.md:123]: all padding
minibatch padding
no padding 

这些要解释下是啥意思

**Review by qingqing01 — APPROVED** [review]
